core/xim/index/IndexStore.lua

```diff
@@ -78,19 +78,30 @@ function _build_index_data(repodir)
     local files = os.files(path.join(repodir, "**.lua"))
     for _, file in ipairs(files) do
         local name = path.basename(file)
-        local triple = string.split(name, "_") -- name_version_maintainer
-        if triple[2] then
-            name = triple[1] .. "@" .. string.gsub(triple[2], "-", ".")
+        local name_maintainer = string.split(name, "_") -- name_version_maintainer
```

Comment:
maintainer is optional in filename. encode maintainer to filename is because we need to distinguish thenm in package index repository.
example:
- openjdk_oracle
- openjdk_zulu
- openjdk_microsoft

Reply:
Shouldn't we encode such info to "factorKey", we only encode the factors that could affect the independence of a package🤔? Like I suggest:
'''lua
-- "factorsKey" = {
--     “versions" = {}
-- }
'''